# Data Types Practice

Short exercises for checking basic type understanding, conversion, and the difference between numbers and strings.

## Problem 1
Create one value for each basic type: `int`, `float`, `str`, and `bool`.
Then print the value and its type.

In [1]:
age = 30
height = 175.5
language = "python"
is_studying = True

print(age, type(age))
print(height, type(height))
print(language, type(language))
print(is_studying, type(is_studying))

30 <class 'int'>
175.5 <class 'float'>
python <class 'str'>
True <class 'bool'>


### Analysis
- `age = 30` creates an integer value.
- `height = 175.5` creates a floating-point value.
- `language = "python"` creates a string value.
- `is_studying = True` creates a boolean value.
- `print(..., type(...))` shows both the value and the type so you can compare them directly.

## Problem 2
A score is stored as text: `"80"`. Convert it to a number and add `20`.

In [2]:
raw_score = "80"
score = int(raw_score)
final_score = score + 20

print(raw_score, type(raw_score))
print(score, type(score))
print(final_score)

80 <class 'str'>
80 <class 'int'>
100


### Analysis
- `raw_score = "80"` stores the score as text, not as a number.
- `score = int(raw_score)` converts the string into an integer so math becomes possible.
- `final_score = score + 20` adds two integers, so Python performs numeric addition.
- The print lines show the type change from `str` to `int`.

## Problem 3
Compare string combination with numeric addition using `"10" + "5"`, `10 + 5`, and `str(10) + "5"`.

In [3]:
text_result = "10" + "5"
number_result = 10 + 5
converted_text_result = str(10) + "5"

print(text_result)
print(number_result)
print(converted_text_result)

105
15
105


### Analysis
- `"10" + "5"` joins two strings, so the result becomes `"105"`.
- `10 + 5` adds two integers, so the result becomes `15`.
- `str(10) + "5"` first turns `10` into text, then joins two strings again.
- This problem shows that the operator looks the same, but the data type changes the behavior.

## Problem 4
Convert the text `"3.14"` into a floating-point number and multiply it by `2`.

In [4]:
raw_pi = "3.14"
pi_value = float(raw_pi)
doubled_value = pi_value * 2

print(raw_pi, type(raw_pi))
print(pi_value, type(pi_value))
print(doubled_value)

3.14 <class 'str'>
3.14 <class 'float'>
6.28


### Analysis
- `raw_pi = "3.14"` starts as a string.
- `pi_value = float(raw_pi)` converts the text into a floating-point number.
- `doubled_value = pi_value * 2` works because `pi_value` is now numeric.
- The output helps you see the change from text to `float`.

## Problem 5
Explain why `"7" + 3` is a problem, then fix it in two different ways.

In [5]:
string_number = "7"
real_number = 3

fixed_as_number = int(string_number) + real_number
fixed_as_text = string_number + str(real_number)

print(fixed_as_number)
print(fixed_as_text)

10
73


### Analysis
- `string_number = "7"` stores the first value as text.
- `real_number = 3` stores the second value as an integer.
- `"7" + 3` would fail because Python does not combine `str` and `int` directly with `+`.
- `fixed_as_number = int(string_number) + real_number` converts both sides into numbers.
- `fixed_as_text = string_number + str(real_number)` converts both sides into strings.
- Both fixes are valid, but they produce different meanings: numeric addition vs text combination.